src/OrleansServiceBus/Providers/Streams/EventHub/CachePressureMonitors/SlowConsumingPressureMonitor.cs

```diff
@@ -0,0 +1,111 @@
+        {
+        }
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="flowControlThreshold"></param>
+        /// <param name="logger"></param>
+        public SlowConsumingPressureMonitor(double flowControlThreshold, Logger logger)
+            : this(flowControlThreshold, DefaultPressureWindowSize, logger)
+        {
+        }
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="flowControlThreshold"></param>
+        /// <param name="pressureWindowSzie"></param>
+        /// <param name="logger"></param>
+        public SlowConsumingPressureMonitor(double flowControlThreshold, TimeSpan pressureWindowSzie, Logger logger)
+        {
+            this.FlowControlThreshold = flowControlThreshold;
+            this.logger = logger.GetSubLogger(this.GetType().Name);
+            this.nextCheckedTime = DateTime.MinValue;
+            this.biggestPressureInCurrentWindow = 0;
+            this.wasUnderPressure = false;
+            this.PressureWindowSize = pressureWindowSzie;
+        }
+
+        public void RecordCachePressureContribution(double cachePressureContribution)
+        {
+            if (cachePressureContribution > this.biggestPressureInCurrentWindow)
+                biggestPressureInCurrentWindow = cachePressureContribution;
+        }
+
+        public bool IsUnderPressure(DateTime utcNow)
+        {
+            //if any pressure contribution in current period is bigger than flowControlThreshold
+            //we see the cache is under pressure
+            bool underPressure = this.biggestPressureInCurrentWindow > this.FlowControlThreshold;
+
+            if (underPressure && !this.wasUnderPressure)
+            {
+                //if under pressure, extend the nextCheckedTime, make sure wasUnderPressure is true for a whole window  
+                this.wasUnderPressure = underPressure;
+                this.nextCheckedTime = utcNow + this.PressureWindowSize;
+                logger.Verbose($"Ingesting messages too fast. Throttling message reading. BiggestPressureInCurrentPeriod: {biggestPressureInCurrentWindow}, Threshold: {FlowControlThreshold}");
+                this.biggestPressureInCurrentWindow = 0;
+            }
+
+            if (this.nextCheckedTime < utcNow)
+            {
+                //at the end of each check period, reset biggestPressureInCurrentPeriod
+                this.nextCheckedTime = utcNow + this.PressureWindowSize;
+                this.biggestPressureInCurrentWindow = 0;
+                //if at the end of the window, pressure clears out, log
+                if(this.wasUnderPressure && !underPressure)
+                    logger.Verbose($"Message ingestion is healthy. BiggestPressureInCurrentPeriod: {biggestPressureInCurrentWindow}, Threshold: {FlowControlThreshold}");
+                this.wasUnderPressure = underPressure;
+            }
+
+            return this.wasUnderPressure;
```

Comment:
in this logic, monitor will stay underPressure for a whole window if a slow consumer appears, while biggestPressureInCurrentWindow will be reset to 0. So at the end of this window, if still underPressure, then underPressure stays for another whole window. Otherwise, monitor become notUnderPressure. 

run through real life case,
monitor will be notUnderPressure at the beginning-> a slow consumer appears ->monitor become underPressure, and stay at underPressure for a whole window -> window ends, if slow consumer cleared out in this window, monitor become notUnderPressure. 

welcome to run through more scenario with me, and let me know if this algorithm is conservative enough